File: src/LYD-CH4/ACW203.cpp
#include <iostream>

using namespace std;

int exgcd(int a, int b, int& x, int& y) {
    if (b == 0) {
        x = 1, y = 0;
        return a;
    } else {
        int d = exgcd(b, a % b, y, x);
        y -= x * (a / b);
        return d;
    }
}

int main() {
    int a,b;
    cin >> a >> b;
    int x,y;
    exgcd(a,b,x,y);
    cout << ((x % b) + b) % b << endl;

    return 0;
}

File: src/LYD-CH4/ACW203.py

def exgcd(a,b,x,y):
    if b == 0:
        x[0] = 1
        y[0] = 0
        return a
    else:
        d = exgcd(b,a % b, y,x)
        y[0] -= x[0] * (a // b)
        return d

s = input().split()
a = int(s[0])
b = int(s[1])

x = [0]
y = [0]

exgcd(a,b,x,y)
x[0] = (x[0] % b + b) % b
print(x[0])
